chia/wallet/did_wallet/did_wallet.py

```diff
@@ -163,6 +172,70 @@ async def create_new_did_wallet_from_recovery(
         self.wallet_id = self.wallet_info.id
         return self
 
+    @staticmethod
+    async def create_new_did_wallet_from_coin_spend(
+        wallet_state_manager: Any,
+        wallet: Wallet,
+        launch_coin: Coin,
+        inner_puzzle: Program,
+        coin_spend: CoinSpend,
+        name: str = "DID Wallet",
+    ):
+        """
+        Create a DID wallet from a transfer
+        :param wallet_state_manager: Wallet state manager
+        :param wallet: Main wallet
+        :param launch_coin: The launch coin of the DID
+        :param inner_puzzle: DID inner puzzle
+        :param coin_spend: DID transfer spend
+        :param name: Wallet name
+        :return:
+        """
+
+        self = DIDWallet()
+        self.base_puzzle_program = None
+        self.base_inner_puzzle_hash = None
+        self.standard_wallet = wallet
+        self.log = logging.getLogger(name if name else __name__)
+        self.wallet_state_manager = wallet_state_manager
+        self.log.info(f"Creating DID wallet from a coin spend {launch_coin}  ...")
+        # Create did info from the coin spend
+        args = did_wallet_puzzles.uncurry_innerpuz(inner_puzzle)
+        if args is None:
+            raise ValueError("Cannot uncurry the DID puzzle.")
+        _, _, num_verification, _, metadata = args
+        full_solution: Program = Program.from_bytes(bytes(coin_spend.solution))
+        inner_solution: Program = full_solution.rest().rest().first()
+        recovery_list: List[bytes] = []
+        for did in list(inner_solution.rest().rest().rest().rest().rest().rest().as_iter()):
+            recovery_list.append(did.as_python()[0])
+        self.did_info = DIDInfo(
+            launch_coin,
+            recovery_list,
+            uint64(num_verification.as_int()),
+            [],
+            inner_puzzle,
+            None,
+            None,
+            None,
+            False,
+            json.dumps(did_wallet_puzzles.program_to_metadata(metadata)),
+        )
+        info_as_string = json.dumps(self.did_info.to_json_dict())
+
+        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
+            name, WalletType.DISTRIBUTED_ID.value, info_as_string, in_transaction=True
```

Comment:
Passing `in_transaction=True` makes sense to me here because `create_new_did_wallet_from_coin_spend()` is called in in the context of an already-opened DB transaction (in `wallet_node.py` `receive_and_validate()`.